### DISCUSSION TITLE
How can i ensure incoming payload is consumed

### DISCUSSION THREAD

**Discussion by rsemihkoca** [emqx/emqx#13475] [General]
Let say i have a topic. I want to make sure that each message sent to this topic consumed rabbitmq like.

How can i do that

**Comment by id** [comment]
You need to publish with QoS 1, monitor the broker to make sure that there is always a subscriber online for that topic, and no messages are dropped.
Subscriber should also use persistent session.

**Comment by zmstone** [comment]
In addition to Ivans answer: EMQX Enterprise is going to support message queue semantics soon. Which does not require a subscriber session created beforehand. It can have messages stored for future subscriptions.
MQTT protocol itself does not have such capacity, only an error code sent back to the publisher (qos1,2) when there is “no match subscriber”.
